packages/framework/esm-styleguide/src/workspaces/workspaces.ts

```diff
@@ -120,6 +124,112 @@ export function canCloseWorkspaceWithoutPrompting(name: string, ignoreChanges: b
   return !promptBeforeFn || !promptBeforeFn();
 }
 
+/**
+ * Closes a workspace group and performs cleanup operations.
+ *
+ * @param groupName - The name of the workspace group to close
+ * @param onWorkspaceCloseup - Optional callback function to execute after closing the workspace group
+ * @returns void, or exits early if the current workspace group name doesn't match the provided group name
+ *
+ * @remarks
+ * This function performs the following operations:
+ * - Validates if the provided group name matches the current workspace group
+ * - Closes all workspaces associated with the group
+ * - Clears the workspace group store state
+ * - Executes cleanup function if defined in the workspace group
+ * - Updates the main workspace store to remove the workspace group
+ * - Calls the optional closeup callback if provided
+ */
+function closeWorkspaceGroup(groupName: string, onWorkspaceCloseup?: Function) {
+  const store = getWorkspaceStore();
+  const currentWorkspaceGroup = store.getState()?.workspaceGroup;
+  const currentGroupName = store.getState()?.workspaceGroup?.name;
+  if (!currentGroupName || groupName !== currentGroupName) {
+    return;
+  }
+  const filter: (workspace: OpenWorkspace) => boolean = currentGroupName
+    ? (workspace) => workspace.currentWorkspaceGroup === currentGroupName
+    : () => true;
+
+  closeAllWorkspaces(() => {
+    // Clearing the workspace group and respective store if the new workspace is not part of the current group, which is handled in the `launchWorkspace` function.
+    const workspaceGroupStore = getWorkspaceGroupStore(groupName);
+    if (workspaceGroupStore) {
+      workspaceGroupStore.setState({}, true);
+      const unsubscribe = workspaceGroupStore.subscribe(() => {});
+      unsubscribe?.();
+    }
+    if (currentWorkspaceGroup && typeof currentWorkspaceGroup?.cleanup === 'function') {
```

Comment:
You really don't need more than this:

```suggestion
    if (typeof currentWorkspaceGroup?.cleanup === 'function') {
```

Reply:
There should also probably be a newline here...